node/primitives/src/lib.rs

```diff
@@ -404,4 +405,37 @@ impl SignedDisputeStatement {
 	pub fn session_index(&self) -> SessionIndex {
 		self.session_index
 	}
+
+	/// Convert a [`SignedFullStatement`] to a [`SignedDisputeStatement`]
+	///
+	/// As [`SignedFullStatement`] contains only the validator index and
+	/// not the validator public key, the public key must be passed as well,
+	/// along with the signing context.
+	///
+	/// This does signature checks again with the data provided.
```

Comment:
Can you elaborate on the `again` in this sentence?

Reply:
I suppose Rob meant that `new_checked` will perform checking of the signature of the dispute statement that we've created here.